src/client_side.cc

```diff
@@ -2154,6 +2154,21 @@ ConnStateData::requestTimeout(const CommTimeoutCbParams &io)
     if (tunnelOnError(HttpRequestMethod(), error))
         return;
 
+    auto doingTls = parsingTlsHandshake; // may be refined below
+#if USE_OPENSSL
+    if (!doingTls) {
+        const auto sslConn = fd_table[io.conn->fd].ssl.get();
+        doingTls = sslConn && !SSL_is_init_finished(sslConn);
+    }
+#else
+    // XXX: Either support GnuTLS (in a regular https_port context) or explain why we cannot.
```

Comment:
If it is not too difficult, please restore parity with GnuTLS here.

Reply:
The commit f52dfebb

More work required for GnuTLS and also probably we need to build a GnuTLS::ErrorDetail class under the GnuTLS namespace.